src/app/inventory/inventory.component.ts

```diff
@@ -13,13 +14,45 @@ import { inventoryActions } from '../store/inventory/inventory.actions';
 export class InventoryComponent {
   entry: MerchandiseEntry;
   inventory$: Observable<MerchandiseEntry[]>;
+  cart$: Observable<MerchandiseEntry[]>;
+  heroAdded: boolean;
+
   constructor(private store: Store<AppState>) {
     this.resetEntry();
     this.inventory$ = store.select('inventory');
+    this.cart$ = store.select('cart');
   }
 
   resetEntry() {
-    this.entry = { merchandise: { displayImage: '', label: '', description: '', price: 0 }, quantity: 0 };
+    this.entry = { merchandise: { displayImage: '', label: '', description: '', price: 0 }, quantity: 0, available: false };
+  }
+
+  addNewHero = () => {
+    const newSuperHero = {
+      quantity: 100,
+      available: true,
+      merchandise: {
+        label: 'Thor',
+        description: 'Thunder storm',
+        price: 29.99,
+        displayImage: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTWkwvBRzcFGPYcPO-AUkJIUcoj-mGK1bPIjvRiWfX_KtcsobGCmA'
+      }
+    };
+    this.store.dispatch(inventoryActions.add(newSuperHero));
+    this.heroAdded = true;
+  }
+
+  removeHero = (index: number, entry: MerchandiseEntry): void => {
+    this.store.dispatch(inventoryActions.remove({index}));
+    this.cart$.subscribe(res => {
```

Comment:
why remove remove should be bothered about the cart? please move it to cart component.